client/components/banner/index.jsx

```diff
@@ -65,14 +66,39 @@ class Banner extends Component {
 	};
 
 	getHref() {
-		const { href, feature, siteSlug } = this.props;
+		const { feature, plan, siteSlug } = this.props;
 
-		if ( ! href && siteSlug ) {
-			if ( feature ) {
-				return `/plans/${ siteSlug }?feature=${ feature }`;
-			}
-			return `/plans/${ siteSlug }`;
+		const baseUrl = `/plans/${ siteSlug }`;
+		let selectedPlan = 'Personal';
+		switch ( plan ) {
+			case 'jetpack_business':
+				selectedPlan = 'Professional';
+				break;
+			case 'jetpack_premium':
+				selectedPlan = 'Premium';
+				break;
+			case 'jetpack_personal':
+				selectedPlan = 'Personal';
+				break;
+			case 'business-bundle':
+				selectedPlan = 'Business';
+				break;
+			case 'value_bundle':
+				selectedPlan = 'Premium';
+				break;
+			case 'personal_bundle':
+				selectedPlan = 'Personal';
+				break;
 		}
+
+		const href = addQueryArgs(
+			{
+				feature: feature,
```

Comment:
If you use object literal property value shorthand, this could be shortened to just:

```
feature,
```

and if you use `plan` instead of `selectedPlan` as I recommended above, it would become:

```
feature,
plan,
```